### DISCUSSION TITLE
Oqtane Dynamic Client Servicing Model

### DISCUSSION THREAD

**Discussion by omer-repo** [oqtane/oqtane.framework#3432] [Q&A]
I discovered Oqtane at ABP Conf and I was very impressed by the Dynamic Client Servicing Model feature.  I have reviewed the presentations about Oqtane many times about this feature but I couldn't find what I was looking for. 

https://www.oqtane.org/blog/!/50/oqtane-3-2-for-net-maui-blazor-hybrid
https://www.oqtane.org/blog/!/51/blazor-client-side-assembly-servicing
https://www.oqtane.org/blog/!/11/assembly-loading-in-blazor-and--net-core
https://github.com/oqtane/oqtane.framework/discussions/2347

What I am looking for is quite simple: Let's imagine that there is a mobile application that I created with Blazor Hybrid (it can be a calculator application). Later we updated a function of this app and we want to update it for users who have already installed it. 

Normally, users need to download it again via Google Play and AppStore. React Native and Expo have this feature. You can host JavaScript files on the server and update the app (without the need to update from the app stores).

It's really great that Blazor Hybrid has this feature as opposed to this feature in Expo. But I couldn't find an article or a video showing how to do it. 

- In order to use this feature in Oqtane, do I need to create the Calculator application with Oqtane? 
- If I do it with Oqtane, do I need to create a module?
- Is there an explanatory document on how to use Oqtane for Blazor Hybrid?
- If I don't have to create the Calculator application in Oqtane for this feature, how can I use the auto-update feature?
- Is there a tutorial that shows an app with this feature from scratch to uploading to the stores?

I apologize in advance if I have missed something despite my research. I'm looking forward to using this feature. Thank you very much for your help.

**Comment by sbwalker** [comment]
The client servicing model is a feature of the Oqtane framework. The blog link https://www.oqtane.org/blog/!/51/blazor-client-side-assembly-servicing contains an overview of how it works - for both Blazor WebAssembly and Blazor Hybrid. In order to leverage this feature you need to build your application as an Oqtane module. The Oqtane framework handles the download of assemblies to the client environment (web, mobile, or desktop). I am not aware of any tutorials other than the presentations I have done which show how to scaffold a new module, compile it, and then run that module (without any code changes) in either a web browser or MAUI container.
